### PR TITLE
task/05-hooks

### PR DESCRIPTION
**Pull request by Allocene**
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Simple Pagination component (without ...) is added and works correctly
- [x]  On the 1st page button Back should be disabled, same as button Forward on the last page
- [x]  Interaction with pagination should not change or break sorting and filtration.
- [x]  Category filter (multi-select) is added and works correctly
- [x]  Sorting input (single select) is added and works correctly
- [x]  Interaction with filter and sorter should reset pagination and move a user to the 1st page

Acceptance Criteria for 1*

- [x]  Pagination component with ... is implemented according to [design](https://www.figma.com/design/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?node-id=1-7667&t=vIBPXrxFdrcCGSep-4).
- [x]  Search input works correctly

https://allocene.github.io/fe-react-2024/

### CODE REVIEW

**Review by Fenix4088 — CHANGES_REQUESTED** [review]
please fix this pagination issues before main. review
https://github.com/Allocene/fe-react-2024/assets/55930010/e691f551-58ad-47b0-9ad2-9727f5a8cade

https://github.com/Allocene/fe-react-2024/assets/55930010/81dbdd78-4d08-4d0b-ab3e-cb530aecc4c1

**Comment by Allocene** [comment]
Added a dependency on changing categories in the Pagination component. In the Products component, reset the current page to the first page when changing the category.

File: src/components/Pagination/Pagination.tsx
Comment (Fenix4088) [inline src/components/Pagination/Pagination.tsx:1]: nice, but move pagination logic into separate hoook
```tsx
const { goToPreviousPage, ....and so on} = usePagination(//some params if u need);
```

File: src/components/Pagination/Pagination.tsx [hunk 1]
```diff
@@ -27,28 +29,85 @@ const Pagination = ({ totalItems, itemsPerPage, onPageChange }: PaginationProps)
...
+
+        return pages;
+    };
 
     return (
         <div className={styles.pagiBox}>
             <ul className={styles.pagination}>
-                <li className={currentPage === 1 ? 'disabled' : ''}>
-                    <Button onClick={goToPreviousPage} className={styles.nonactive}>
+                <li className={currentPage === 1 ? styles.disabled : ''}>
```
Comment (Fenix4088) [inline src/components/Pagination/Pagination.tsx]: DRY- don't repeat yourself
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself

```ts
const isFirstPage = currentPage === 1;
```

File: src/components/Products/Products.tsx [hunk 2]
```diff
@@ -14,25 +14,76 @@ interface ProductsProps {
     isDarkTheme: boolean;
 }
 
+const filterProducts = (products: typeof mock, filter: string, searchQuery: string, selectedCategories: string[]) => {
```
Comment (Fenix4088) [inline src/components/Products/Products.tsx]: create a type instead of typeof mock

File: src/components/Products/Products.tsx [hunk 3]
```diff
@@ -14,25 +14,76 @@ interface ProductsProps {
...
+    let filteredProducts = products;
```
Comment (Fenix4088) [inline src/components/Products/Products.tsx]: why? filteredProducts i redundant

File: src/components/Products/Products.tsx [hunk 4]
```diff
@@ -14,25 +14,76 @@ interface ProductsProps {
...
+    switch (filter) {
+        case 'price': {
+            return filteredProducts.sort((a, b) => b.price - a.price);
+        }
+        case 'newest': {
+            return filteredProducts.sort((a, b) => new Date(b.creationAt).getTime() - new Date(a.creationAt).getTime());
+        }
+        case 'oldest': {
+            return filteredProducts.sort((a, b) => new Date(a.creationAt).getTime() - new Date(b.creationAt).getTime());
+        }
```
Comment (Fenix4088) [inline src/components/Products/Products.tsx]: price, newest, oldest 
create enum,

File: src/components/Products/Products.tsx [hunk 5]
```diff
@@ -14,25 +14,76 @@ interface ProductsProps {
     isDarkTheme: boolean;
 }
 
+const filterProducts = (products: typeof mock, filter: string, searchQuery: string, selectedCategories: string[]) => {
+    let filteredProducts = products;
+
+    if (searchQuery) {
```
Comment (Fenix4088) [inline src/components/Products/Products.tsx]: move logic related with filtering into separate custom hookk

**Review by Fenix4088 — CHANGES_REQUESTED** [review]

File: src/components/Pagination/Pagination.tsx
Comment (Allocene) [reply src/components/Pagination/Pagination.tsx:1]: Done

**Review by Fenix4088 — APPROVED** [review]

**Pull request merged**